knowage-vue/src/modules/documentExecution/main/dialogs/documentExecutionMetadataDialog/DocumentExecutionMetadataDialog.vue

```diff
@@ -0,0 +1,195 @@
+        propLoading() {
+            this.setLoading()
+        }
+    },
+    computed: {
+        canModify(): boolean {
+            return (this.$store.state as any).user.functionalities.includes('SaveMetadataFunctionality')
+        }
+    },
+    created() {
+        this.setLoading()
+        this.loadDocument()
+        this.loadMetadata()
+    },
+    methods: {
+        loadDocument() {
+            this.document = this.propDocument
+        },
+        loadMetadata() {
+            if (this.propMetadata) {
+                this.metadata = {
+                    generalMetadata: this.propMetadata.GENERAL_META ? [...this.propMetadata.GENERAL_META] : [],
+                    shortText: this.propMetadata.SHORT_TEXT ? [...this.propMetadata.SHORT_TEXT] : [],
+                    longText: this.propMetadata.LONG_TEXT ? [...this.propMetadata.LONG_TEXT] : [],
+                    file: this.propMetadata.FILE ? [...this.propMetadata.FILE] : []
+                }
+            }
+        },
+        setLoading() {
+            this.loading = this.propLoading
+        },
+        uploadFile(event: any) {
+            this.uploadedFiles[event.target.id] = event.target.files[0]
+        },
+        async uploadMetaFile(meta: any) {
+            if (!this.uploadedFiles[meta.id]) {
+                this.$store.commit('setError', {
+                    title: this.$t('common.error.generic'),
+                    msg: this.$t('documentExecution.main.selectFileError')
+                })
+            } else {
+                this.loading = true
+
+                const formData = new FormData()
+                formData.append('file', this.uploadedFiles[meta.id])
+                formData.append('fileName', this.uploadedFiles[meta.id].name)
+                await this.$http
+                    .post(process.env.VUE_APP_RESTFUL_SERVICES_PATH + `1.0/documentexecution/uploadfilemetadata`, formData, { headers: { 'Content-Type': 'multipart/form-data' } })
+                    .then(() => {
+                        this.$store.commit('setInfo', {
+                            title: this.$t('common.uploadFile'),
+                            msg: this.$t('common.uploadFileSuccess')
+                        })
+                    })
+                    .catch((error: any) =>
+                        this.$store.commit('setError', {
+                            title: this.$t('common.error.generic'),
+                            msg: error
+                        })
+                    )
+                this.loading = false
+            }
+        },
+        cleanFile(meta: any) {
+            const temp = this.$refs[meta.id] as any
+            if (temp) {
+                temp.resetInput()
+            }
+        },
+        closeDialog() {
+            this.metadata = null
+            this.$emit('close')
+        },
+        save() {
+            this.$emit('saveMetadata', this.metadata)
+        }
+    }
+})
+</script>
+
+<style lang="scss">
```

Comment:
put scoped please if not needed for children, the pi-upload class is too generic

Reply:
Ok, leave it, I will take care of the class to add.